### PR TITLE
(fix) Trivia Page

### PR DESCRIPTION
**Pull request by mazenchami**
# Description

[Trello Card](165-day-2-after-lunch-talk-should-be-trivia-game) — #140 

Making `Trivia Show` card clickable and display like a multi-speaker talk page

# Graphics

| iOS            | Android            |
| -------------- | ------------------ |
|  |  |

# Checklist:

- [ ] I have done a thorough self-review of my code
- [ ] I have tested with a screen reader and font-scaling turned on and added necessary accessibility features
- [ ] I have run tests and linter

### CODE REVIEW

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx [hunk 1]
```diff
@@ -60,46 +71,77 @@ export interface TalkDetailsProps {
...
+        { url: speaker?.externalURL || speaker?.website, icon: "link" },
+      ],
+    })),
+  }
+}
+
 const talkDetailsProps = (schedule: ScheduledEvent): TalkDetailsProps => {
   const talk = schedule.talk
 
+  if (schedule["event-title"] === WEBFLOW_MAP.triviaShow.title) return triviaShowProps(schedule)
```
Comment (morganick) [inline app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:132]: The conditionals will continue until morale improves!

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx [hunk 2]
```diff
@@ -115,12 +157,76 @@ const talkDetailsProps = (schedule: ScheduledEvent): TalkDetailsProps => {
       socialButtons: [
         { url: speaker?.twitter, icon: "twitter" },
         { url: speaker?.github, icon: "github" },
-        { url: speaker?.externalURL, icon: "link" },
+        { url: speaker?.externalURL || speaker?.website, icon: "link" },
```
Comment (frankcalise) [inline app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:160]: multiple webflow field names for the same thing rip :(

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx (hunk 1 above)
Comment (frankcalise) [inline app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:132]: do we go contains here for the day of title change? hoping no other talk has trivia in it?

a fun roll of the dice for sure

**Review by frankcalise — APPROVED** [review]
looks pretty straight forward, like the refactors for single/multi speaker!

hardcody feel to it but is what it is I think

File: app/services/api/webflow-api.types.ts [hunk 3]
```diff
@@ -130,6 +130,7 @@ export interface RawSpeaker extends Item {
   title?: string
   twitter?: string
   externalURL?: string
+  website?: string
```
Comment (morganick) [inline app/services/api/webflow-api.types.ts:133]: I'm guessing this was an addition on the webflow side?

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx [hunk 4]
```diff
@@ -182,41 +274,16 @@ export const TalkDetailsScreen: FC<StackScreenProps<AppStackParamList, "TalkDeta
...
+            {isMultipleSpeakers ? (
+              <TalkDetailsMultipleSpeakers
+                {...(talkDetailsProps(schedule) as TalkDetailsMultipleSpeakersProps)}
+                scheduleType={schedule.type}
+              />
+            ) : (
+              <TalkDetailsSingleSpeaker
+                {...(talkDetailsProps(schedule) as TalkDetailsSingleSpeakerProps)}
+                scheduleType={schedule.type}
+              />
```
Comment (morganick) [inline app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:286]: 👏 Thanks the conditionals were getting hard to parse.

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx (hunk 1 above)
Comment (morganick) [reply app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:132]: Is this title going to change? If so this wouldn't be the only place we'd have logic changes.

**Review by morganick — CHANGES_REQUESTED** [review]
Minor tweaks then g2g.

File: app/services/api/webflow-api.types.ts (hunk 3 above)
Comment (mazenchami) [reply app/services/api/webflow-api.types.ts:133]: yes. we have a miro board that has all the fields as they're being add but this was missed

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx [hunk 5]
```diff
@@ -115,12 +157,76 @@ const talkDetailsProps = (schedule: ScheduledEvent): TalkDetailsProps => {
...
       ],
     })),
   }
 }
 
+type TalkDetailsSingleSpeakerScreenProps = TalkDetailsSingleSpeakerProps & {
+  scheduleType: ScheduledEvent["type"]
+}
+
+const TalkDetailsSingleSpeaker: React.FunctionComponent<TalkDetailsSingleSpeakerScreenProps> =
```
[the remark this answers is not in the document]
Comment (mazenchami) [reply app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:170]: either or. `FC` just maps to `FunctionComponent`:

Screenshot 2023-04-21 at 2 35 25 PM

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx [hunk 6]
```diff
@@ -60,46 +71,77 @@ export interface TalkDetailsProps {
...
    */
-  isMultipleSpeakers: boolean
-  /**
-   * The carousel data
-   */
-  carouselData?: Array<DynamicCarouselItem>
+  isMultipleSpeakers: false
 }
 
+export type TalkDetailsProps = TalkDetailsMultipleSpeakersProps | TalkDetailsSingleSpeakerProps
```
[the remark this answers is not in the document]
Comment (mazenchami) [reply app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx]: updated here: https://github.com/infinitered/ChainReactApp2023/pull/156/commits/d0ce4dc40e2ea0252087fbb84771d5de8721124b

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx (hunk 1 above)
Comment (mazenchami) [reply app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:132]: yeah i'd a contains might bite us considering how sensitive this is right now

**Comment by mazenchami** [comment]
@morganick regarding: https://github.com/infinitered/ChainReactApp2023/pull/156#discussion_r1174048208:

My goal is that using the `WEBFLOW_MAP.triviaShow` const would help us where we would only have to change it in one spot (consts file) and push an OTA update to solve this issue quickly

File: app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (morganick) [reply app/screens/TalkDetailsScreen/TalkDetailsScreen.tsx:170]: Yeah, I just mean for consistency.

**Review by morganick — APPROVED** [review]
